addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -1028,6 +1028,51 @@ registry.HeaderHamburgerFull = publicWidget.Widget.extend({
     },
 });
 
+registry.BottomFixedElement = publicWidget.Widget.extend({
+    selector: '#wrapwrap',
+
+    /**
+     * @override
+     */
+    start() {
+        this.$scrollingElement = $().getScrollingElement();
+        this.$scrollingElement.on('scroll.bottom_fixed_element', _.debounce(() => this._hideBottomFixedElements(), 500));
+        $(window).on('resize.bottom_fixed_element', _.debounce(() => this._hideBottomFixedElements(), 500));
+        return this._super(...arguments);
+    },
+    /**
+     * @override
+     */
+    destroy() {
+        this._super(...arguments);
+        this.$scrollingElement.off('.bottom_fixed_element');
+        $(window).off('.bottom_fixed_element');
+        $('.o_bottom_fixed_element').removeClass('o_bottom_fixed_element_hidden');
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Hide the element that is fixed at the bottom of the screen if the scroll
+     * reaches the bottom of the page and if this element hides a button.
+     *
+     * @private
+     */
+    _hideBottomFixedElements() {
+        const $bottomFixedElements = $('.o_bottom_fixed_element');
+        if ($bottomFixedElements.length) {
+            const isAtBottom = ((this.$scrollingElement[0].offsetHeight + this.$scrollingElement[0].scrollTop) >= (this.$scrollingElement[0].scrollHeight - 2));
+            const $buttons = this.$target.find('.btn:visible');
+            $bottomFixedElements.removeClass('o_bottom_fixed_element_hidden');
```

Comment:
The point of the next code was also to get rid of this line

Reply:
Here I need to remove the class to avoid to have display: none of elements before to check getBoundingClientRect() of it. Do you see an other solution ?

Reply:
Mmh indeed... but in that case the use of toggle below loses its meaning. You can go back to a "add" and the "isAtBottom" condition can be put around the whole `for` loop.